scripts/run_scrnaseq_hto.R

```diff
@@ -0,0 +1,185 @@
+# Param project id
+paramsList[["project_id"]] = as.character(opt[["project_id"]])
+
+# Param path_data
+if (!"path_data" %in% names(opt) || is.null(opt[["path_data"]])) stop("Need to provide the path to a filtered counts directory created by the 10X cellranger pipeline via --path-data!")
+if (!file.exists(opt[["path_data"]])) stop("Path to a filtered counts directory specified via --path-data does not exists or is not a directory!")
+paramsList[["path_data"]] = normalizePath(opt[["path_data"]])
+
+# Param stats
+if ("stats" %in% names(opt) && !is.null(opt[["stats"]])) {
+    if (!file.exists(opt[["stats"]])) stop("Path to a metrics_summary.csv file specified via --stats does not exists or is not a file!")
+    paramsList[["stats"]] = normalizePath(opt[["stats"]])
+} else {
+    paramsList["stats"] = list(NULL)
+}
+
+# Param path_out
+if (!file.exists(opt[["path_out"]])) dir.create(opt[["path_out"]])
+paramsList[["path_out"]] = normalizePath(opt[["path_out"]])
+
+# Param hto-features
+if ("hto_names" %in% names(opt) && !is.null(opt[["hto_names"]])) {
+    hto_names_split = strsplit(trimws(unlist(strsplit(opt[["hto_names"]], ","))),"=")
+    hto_names_n = sapply(hto_names_split, function(h) {return(h[1])})
+    hto_names_s = sapply(hto_names_split, function(h) { if(length(h)==2) return(h[2]) else return(h[1])})
+    
+    paramsList[["hto_names"]] = setNames(hto_names_s, hto_names_n)
+} else {
+    paramsList["hto_names"] = list(NULL)
+}
+
+# Param hto-features-regex
+if ("hto_regex" %in% names(opt) && !is.null(opt[["hto_regex"]])) {
+    if (!is.null(paramsList[["hto_names"]])) stop("Cannot use --hto-features-regex together with --hto-features!")
+    paramsList[["hto_regex"]] = trimws(opt[["hto_regex"]])
+} else {
+    paramsList["hto_regex"] = list(NULL)
+}
+
+if (is.null(paramsList[["hto_regex"]]) & is.null(paramsList[["hto_names"]])) stop("Need to specify HTO features for sample demultiplexing either via --hto-features or via --hto-features-regex!")
+
+# Param norm
+if ("norm" %in% names(opt)) {
+    if (!opt[["norm"]] %in% c("CLR","LogNormalize")) stop("Only 'CLR' or 'LogNormalize' allowed for --norm!")
+    paramsList[["norm"]] = opt[["norm"]]
+}
+
+# Param mt_names
+if ("mt_names" %in% names(opt)) {
+    paramsList[["mt_names"]] = opt[["mt_names"]]
+}
+
+# Param col
+if ("col" %in% names(opt)) {
+    if (!opt[["col"]] %in% colours()) stop("Only R colours allowed for --col!")
+    paramsList[["col"]] = opt[["col"]]
+}
+
+# Param downsample_cells_n
+if ("downsample_cells_n" %in% names(opt) && !is.null(opt[["downsample_cells_n"]])) {
+    paramsList[["downsample_cells_n"]] = opt[["downsample_cells_n"]]
+}
+
+# Param path_to_git
+if ("path_to_git" %in% names(opt)) {
+    paramsList[["path_to_git"]] = opt[["path_to_git"]]
+}
+
+# Param debugging_mode: 'default_debugging' for RStudio, 'terminal_debugger' for debugging on a terminal without X11, 'print_traceback' for non-interactive sessions
+paramsList[["debugging_mode"]] = 'print_traceback'
+
```

Comment:
Well in this case it may indeed be a bit redundant since if not set it will be set automatically in the rmd script.

The wrapper script is neccessary to safely and nicely call the rmd script.

Reply:
Yes I get it. But in the case of the main script, you do all the checks within the Rmd. It's fine, just a bit inconsistent.